### ISSUE TITLE
[Tracking Issue] Implement Integraion Tests

### ISSUE THREAD

**Issue by YJDoc2** [youki-dev/youki#361]
As per #56 , we are porting the OCI validation tests into rust. This is the tracking issue for the same.
The folder /youki_integration_test contains the currently implemented tests which are lifecyle, creation and huge_tlb

Here is the documentation for developing the integration tests:
https://youki-dev.github.io/youki/developer/e2e/rust_oci_test.html

**Please only work at one test at a time, and wait for its PR to get merged before opening another one**. It helps us to have an easier way for reviewing the PRs 🙏 

## Default
List of all test, as per https://github.com/opencontainers/runtime-tools/tree/master/validation :
- [ ] config_updates_without_affect
  - Original test code in Go: https://github.com/opencontainers/runtime-tools/blob/master/validation/config_updates_without_affect/config_updates_without_affect.go
- [ ] create(@sivchari )
  - Original test code in Go: https://github.com/opencontainers/runtime-tools/blob/master/validation/create/create.go
- [ ] default
  - Original test code in Go: https://github.com/opencontainers/runtime-tools/blob/master/validation/default/default.go
- [x] delete (@sou1118 ) https://github.com/youki-dev/youki/pull/3082
- [ ] delete_only_create_resources
- [ ] delete_resources
- [x] hooks
- [ ] hooks_stdin
- [x] hostname(@chermehdi)
- [x] kill (@YamasouA  https://github.com/youki-dev/youki/pull/2996)
- [ ] kill_no_effect
- [ ] killsig (@YamasouA )
- [x] linux_cgroups_blkio(#537)
- [x] linux_cgroups_cpus(#462)
- [ ] linux_cgroups_devices ( @sat0ken )
- [x] linux_cgroups_hugetlb
- [x] linux_cgroups_memory(#473)
- [x] linux_cgroups_network(#516)
- [x] linux_cgroups_pids(#391)
- [ ] linux_cgroups_relative_blkio(@furon-kuina)
- [ ] linux_cgroups_relative_cpus // *Please take over from https://github.com/youki-dev/youki/pull/2898*
- [ ] linux_cgroups_relative_devices (@sat0ken )
- [ ] linux_cgroups_relative_hugetlb(@kmpzr)
- [ ] linux_cgroups_relative_memory(@omprakaash)
- [x] linux_cgroups_relative_network (@moz-sec  https://github.com/youki-dev/youki/pull/2986 )
- [ ] linux_cgroups_relative_pids(@kmpzr)
- [x] linux_devices https://github.com/containers/youki/pull/2708 by @omprakaash
- [x] linux_masked_paths (@nayuta-ai https://github.com/youki-dev/youki/pull/2950)
- [ ] linux_mount_label @Gekko0114 
- [x] linux_ns_itype(#389)
- [ ] linux_ns_nopath
- [ ] linux_ns_path
- [ ] linux_ns_path_type
- [ ] linux_process_apparmor_profile
- [x] linux_readonly_paths (#582 )
- [x] linux_rootfs_propagation (@saku3 https://github.com/youki-dev/youki/pull/3024 )
- [x] linux_seccomp https://github.com/containers/youki/pull/2531 by @xiaoyang-sde 
- [x] linux_sysctl https://github.com/containers/youki/pull/2527 by @xiaoyang-sde 
- [x] linux_uid_mappings (@moz-sec https://github.com/youki-dev/youki/pull/3161)
- [ ] misc_props(@YamasouA https://github.com/youki-dev/youki/pull/3250)
- [ ] mounts(@you-matsuura)
- [x] pidfile(#379)
- [ ] poststart
- [ ] poststart_fail
- [ ] poststop
- [ ] poststop_fail
- [ ] prestart
- [ ] prestart_fail
- [x] process ( @sat0ken https://github.com/youki-dev/youki/pull/2968 )
- [ ] process_capabilities
- [x] process_capabilities_fail (@kazmsk https://github.com/youki-dev/youki/pull/3010)
- [x] process_oom_score_adj ( @saku3 https://github.com/youki-dev/youki/pull/2987 )
- [x] process_rlimits ( @sat0ken https://github.com/youki-dev/youki/pull/2977)
- [x] process_rlimits_fail ( @ntkm61027 https://github.com/youki-dev/youki/pull/3051)
- [x] process_user ( @sat0ken https://github.com/youki-dev/youki/pull/2978)
- [x] root_readonly_true ( @sat0ken https://github.com/youki-dev/youki/pull/2976)
- [ ] start
- [ ] state

## Extensions
This contains tests are not in https://github.com/opencontainers/runtime-tools/tree/master/validation , or need are not satisfactory enough.
- [ ] cgroups v1
- [ ] cgroups v2

Out of these, currently only huge_tlb is ported completely. lifecycle and create tests are implemented but need much improvements, and lifecycle tests should be split into individuals as per listed above.
Please read the README of youki_integration_test to get idea for how one should go about implementing the tests, and what utils etc are available.

**Comment by YJDoc2** [comment]
Waiting for #360 , which moves test_framework into youki_integration_test, but other than that , ready for individual test claiming.

**Comment by utam0k** [comment]
@YJDoc2 That's great! It would be a good idea to add a link to the sample implementation you implemented to the description of this issue.

**Comment by tsturzl** [comment]
I'll have some free time soon and I'd love to pitch in on this effort! I think this tooling will be a force multiplier and I think it's really impactful. Are there any pain points that I could tackle that anyone can think of that might not be a good first issue? Otherwise I'm more than happy to just jump in on cgroups controllers where I'm pretty familiar.

**Comment by YJDoc2** [comment]
Hey @tsturzl , sorry for late reply. Currently there are no obvious pain points that I know, and as you are already familiar with cgroups, it'd be great if you can work on those tests. Also if you take any test from the first list (which oci-tools have) please comment here, so we can keep track.

@utam0k  I am a bit busy for few days, but after that I'm planning to work on either pid test or ns_itype test as currently both fail for runc and youki using oci-tests. Assuming that the issue is in test implementation, and not in the runtime, if we could support those tests, we can theoretically be feature equivalent with oci-tests.

**Comment by utam0k** [comment]
> @utam0k i want to implement "config_updates_without_affect"

@WormOn 
Sure. I assigned you.

**Comment by tsturzl** [comment]
I'll start off with `linux_cgroups_cpus`, I think it will be an easy one to start off with.

**Comment by tsturzl** [comment]
I'm nearly done with the linux_cgroups_cpus integrations tests. I'd like to take on the memory cgroups tasks. I'd really like to fix a problem that currently exists in the current oci validations with these specific tests.

**Comment by tsturzl** [comment]
Also, maybe we should limit the context of this tracking issue to cgroups v1 and just generally getting this on par with oci runtime-tools? Perhaps then we can open another tracking issue to detail out what cgroups v2 tests should look like and what needs to get implemented.

**Comment by YJDoc2** [comment]
Hey @tsturzl , sure you can take memory next :smile: , I'll assign it to you then.
It is a good idea that we use this only for tracking migration, and open separate one for cgroup v2, but I feel it'd be good to mention cgroups v2 here, until we have opened that issue, then we can link that here.

**Comment by YJDoc2** [comment]
Hey @WormOn, how is the implementation going? Let know if you need any help in working on the test implementation :+1:

**Comment by tsturzl** [comment]
I got bored and started working on linux_cgroups_network, so I'll take that one also

**Comment by whereistejas** [comment]
Hi! I would like to contribute to this issue by adding a few tests, too. I have been working with rust for about 6 months, now. However, I have no exp with container runtimes, thus, would like to start with small and low priority tests.

I though I could start by adding the `root_readonly_true` test? Any thoughts?

**Comment by YJDoc2** [comment]
Hey @whereistejas , thanks for your interest in contribution. I would suggest that rather than starting with `root_readonly_test` , you might start with `craete`. The reason behind this is that `root_readonly` uses a function `test_inside_runtime` a util function, which I am currently implementing for another test, but facing some issues. I hope to get it worked out in some days.

The create test uses the basic functions which are already implemented in the current utils, you can check the lifecycle tests to get an idea how to implement the create tests. If you need any help or are facing some issues, you can ask them here or on the youki discord.

Thanks :)

**Comment by whereistejas** [comment]
Hi @YJDoc2, if my understanding is correct you are referring to the following [`create`](https://github.com/opencontainers/runtime-tools/blob/master/validation/create/create.go) tests. I can see these tests are already implemented as part of `lifecycle/container_create.rs`. Please, correct me if I'm wrong.

**Comment by YJDoc2** [comment]
Hey, yes that are the tests, but they were created as part of proof-of-concept of integration tests themselves, and should be replaced in sometime for their own dedicated tests as per the runtime tools.

Or if you are fine with it, you can wait till merging of #582 which implemented the `test_inside_container` , then you can work on `root_readonly` as well.

**Comment by whereistejas** [comment]
> and should be replaced in sometime for their own dedicated tests as per the runtime tools.

So, we would like to have the `create` tests extracted/refactored out of `lifecycle` into their own module?

**Comment by YJDoc2** [comment]
Hey @whereistejas , yes I meant that create tests should be refactored in there own module, as you are working on in #583  :+1:
Thank you for contributing!

**Comment by YJDoc2** [comment]
Hey there was a pause recently on adding the tests as we were refactoring the directory structure, which is now completed! So if anyone is interested, you can work on adding the new tests and send PRs. Thank you :)

ping @Szymongib if you are still interested :sweat_smile:

**Comment by chermehdi** [comment]
I'll try to tackle a couple as well

**Comment by utam0k** [comment]
> I'll try to tackle a couple as well

@chermehdi cc: @YJDoc2 
Which test cases do you have interested in? If you can choose, I will assign you ;)

**Comment by chermehdi** [comment]
Currently still taking a look at the runc tests, just to make sure we are on the same page, we are talking about porting these [tests](https://github.com/opencontainers/runc/tree/main/tests/integration) written in bat to Youki's own testing framework?

**Comment by utam0k** [comment]
> Currently still taking a look at the runc tests, just to make sure we are on the same page, we are talking about porting these [tests](https://github.com/opencontainers/runc/tree/main/tests/integration) written in bat to Youki's own testing framework?

@chermehdi 
In our case, we are writing in our own framework, although it is a bit more challenging so that we can be more flexible.
But first, we want to port the [runtime-tools test cases.](https://github.com/opencontainers/runtime-tools/tree/master/validation) Unfortunately, the maintainer is not active and is even less flexible.

**Comment by chermehdi** [comment]
Aaa I see, let me take the create (first one I saw here), and once we have that, let's see how it goes from there, how does that sound?

**Comment by utam0k** [comment]
Maybe a port of hostname to get into our development flow for starters?
https://github.com/opencontainers/runtime-tools/blob/master/validation/hostname/hostname.go

We're still in our infancy, so we'd love to hear about any improvements you've found unsatisfactory in the process!

**Comment by chermehdi** [comment]
sure, sounds good 👍

**Comment by utam0k** [comment]
@chermehdi I assigned you. Thanks :pray:

**Comment by xiaoyang-sde** [comment]
Hi! I'm interested in contributing to this project and I guess this issue could be a good starting point. I'd love to take on some of the tasks and see how I can assist.

**Comment by utam0k** [comment]
@xiaoyang-sde Thanks for your comment. I'm glad to hear that.

**Comment by utam0k** [comment]
@YJDoc2 May I ask you to update this tasks in this issue?

**Comment by utam0k** [comment]
@xiaoyang-sde Is there any task you are interested in?

**Comment by xiaoyang-sde** [comment]
> @xiaoyang-sde Is there any task you are interested in?

I'm still trying to understand the source code. I will let you know soon!

**Comment by YJDoc2** [comment]
> @YJDoc2 May I ask you to update tasks in this issue?

Hey, I'll take a look and update them, thanks :+1:

>I'm still trying to understand the source code. I will let you know soon!

@xiaoyang-sde feel free to ping if you need any help

**Comment by utam0k** [comment]
Thanks @YJDoc2 😍

**Comment by MarceloSpessoto** [comment]
Hello! Can I tackle the `linux_cgroups_devices` test?

**Comment by tsturzl** [comment]
@MarceloSpessoto Devices is really only feature complete for cgroups v1 right now, so for cgroups v2 the test might need to be skipped. It may be hard to validate the test on a cgroups v2 system, and might require testing in a VM. If your comfortable with that I'm happy to assign you. Otherwise any of the other remaining `linux_cgroups_*` tests would be a great place to start!

Unfortunately cgroups v2 doesn't really have a mechanism for managing devices, and the workaround is [fairly complicated](https://github.com/containers/youki/issues/230).

**Comment by MarceloSpessoto** [comment]
Hi @tsturzl , I didn't know about these cgroups v2 caveats. I was looking for something easier to start with, so I think I prefer changing to `linux_cgroups_relative_blkio`.

**Comment by tsturzl** [comment]
@MarceloSpessoto Assigned! Feel free to reach out if you need help.

**Comment by utam0k** [comment]
@tsturzl Thanks for your handling and guiding 🙏

**Comment by kmpzr** [comment]
Hello :wave:
I'm with some capacity nowadays and would be up to tackle a thing or two. Since I previously touched [PID](https://github.com/containers/youki/pull/26) & [hugetlb](https://github.com/containers/youki/pull/135) I could pick  `linux_cgroups_relative_hugetlb` and `linux_cgroups_relative_pids`. But I'm fine with anything you assign to me given your priority :smile:

**Comment by omprakaash** [comment]
Hello ! Can I take up `linux_cgroups_relative_memory` ?

**Comment by tsturzl** [comment]
@kmpzr assigned those two to you. Let me know if you have any questions!

**Comment by tsturzl** [comment]
@omprakaash assigned! Let me know if you have questions or need help. I'm actually the original author of the v1 and v2 cgroups memory controller, so I'm pretty familiar with that code base.

**Comment by utam0k** [comment]
@kmpzr @omprakaash I'm looking forward to seeing your PR!

**Comment by omprakaash** [comment]
> @omprakaash assigned! Let me know if you have questions or need help. I'm actually the original author of the v1 and v2 cgroups memory controller, so I'm pretty familiar with that code base.

Thank You. Will do !

**Comment by Gekko0114** [comment]
Hi @utam0k,
Can I work on `linux_mount_label` ?

**Comment by utam0k** [comment]
> Hi @utam0k, Can I work on `linux_mount_label` ?

I'm happy to hear that. I assigned you.

**Comment by omprakaash** [comment]
Hey, can I take on `linux_devices`?

**Comment by utam0k** [comment]
@furon-kuina Sure, a good challenge! I appreciate your challenge. I'll assign you to `linux_cgroups_relative_blkio`.  I guess there already exists a similar implemented test case. I suggest referring them.

**Comment by furon-kuina** [comment]
@utam0k 
Thank you so much! I'll take a look at it!

**Comment by YJDoc2** [comment]
Hey @furon-kuina , check https://github.com/containers/youki/pull/2722 once, which was opened for the linux_cgroups_relative_blkio test, but the author could not continue. You can use that as a starting point or reference to get an idea for how to implement the test. (that said, the PR has several issues, failing CI etc so you might need to fix / rewrite it)

**Comment by utam0k** [comment]
> Hey @furon-kuina , check #2722 once, which was opened for the linux_cgroups_relative_blkio test, but the author could not continue. You can use that as a starting point or reference to get an idea for how to implement the test. (that said, the PR has several issues, failing CI etc so you might need to fix / rewrite it)

Ops, thanks for your follow-up 🙏

**Comment by furon-kuina** [comment]
@YJDoc2 @utam0k 
Thanks for the help! I really appreciate it.
I'll study the youki User and Developer Documentation and get started with the implementation.

**Comment by posutsai** [comment]
Hi @utam0k, I would like to take the `linux_cgroups_relative_cpus`. Please assign me if possible.

**Comment by YJDoc2** [comment]
@posutsai assigned to you :+1:

**Comment by posutsai** [comment]
Hi @YJDoc2 and @utam0k,
After a few day survey, seems like testing `linux_cgroups_relative_cpus` relates to the API of cgroup v1, referring the [comparison table](https://www.alibabacloud.com/help/en/alinux/support/differences-between-cgroup-v1-and-cgroup-v2).  Given my Ubuntu is using cgroup v2, Is it possible to make cgroup v1 and v2 coexist? or is there any tool to simulate the behavior? Thank you.

**Comment by YJDoc2** [comment]
Hey, so as far as I know, recent Ubuntu versions have dropped support of cgroups v1 by default, you need to do certain kernel feature enabling to enable cgroups v1 on them. I think ubuntu 20.04 has cgroups v1 enabled in them, and we use that in our ci -check https://github.com/containers/youki/pull/2645 . An easier way for your testing can be to create a vagrant box with this version, mount the youki dir in it, compile your things on host systems, so you don't have to install rust and build deps in the box, but run the tests in the box instead of host (so you'll need the runtime deps). 

I have also opened https://github.com/containers/youki/issues/2890 for standardizing the test envs via vagrant boxes.

**Comment by utam0k** [comment]
> Hi! This is my first time with youki. I am interested in contributing to the youki project! I would like to know if it would be possible to assign me to the `linux_masked_paths`.

Sure! I've assigned you. Thanks!

**Comment by utam0k** [comment]
> Hi! This is my first time with youki. I am interested in contributing to the youki project! I would like to know if it would be possible to assign me to the `linux_masked_paths`.

@nayuta-ai Sure! I've assigned you. Thanks!

**Comment by sou1118** [comment]
Hi @utam0k and @YJDoc2
I want to work on `delete` job. Could you assign me? This is my first OSS and Youki contribution.

**Comment by YJDoc2** [comment]
Hey @sou1118 , I have assigned you the `delete` test, just check once if it has already been implemented or not (it is possible that we might have missed crossing it off the list). Also welcome to Youki, and OSS as whole! Hope you'll have fun and find the contributing interesting. Ping me if you need any help!

**Comment by YamasouA** [comment]
@YJDoc2 @utam0k
Hi.
I'm intereseted in this Issue.
Can you assign any of the test cases to me?

**Comment by YJDoc2** [comment]
@moz-sec , assigned the `linux_cgroups_relative_network` to you. Looking forward to you PR!

@YamasouA , I have assigned the `kill` test to you, please take a look. If you want some other let me know and I can re-assign.

In case either of you need any help while working on this, please feel free to tag/ping me here :)

**Comment by saku3** [comment]
@YJDoc2 @utam0k 
Hi!
I would like to be assigned to `process_oom_score_adj`. Could you please assign it to me?

**Comment by YJDoc2** [comment]
@saku3 done! Feel free to ping me if you have any issues while working on this :+1:

**Comment by sat0ken** [comment]
@YJDoc2 
Hi ! I wanna try to linux_cgroups_devices and linux_cgroups_relative_devices. Could you please assign it to me?

**Comment by saku3** [comment]
@YJDoc2
Hi ! I wanna try to linux_rootfs_propagation. Could you please assign it to me?

**Comment by moz-sec** [comment]
Hi @YJDoc2 
I would like to be assigned to `linux_uid_mappings`
Could you please assign it to me?

**Comment by kazmsk** [comment]
Hi @YJDoc2
I would like to be assigned to `process_capabilities_fail`.
Could you please assign it to me?

**Comment by YJDoc2** [comment]
@saku3 assigned linux_rootfs_propagation
@moz-sec , please wait till your existing PR is merged 😅 I think there are still some changes I need to review. However, I'll assign this to you once this is merged.
@kazmsk , assigned process_capabilities_fail.

Also, anyone who has an open PR of e2e testing, please rebase your PR, as we have just merged a big refactor of one of the functions in  https://github.com/youki-dev/youki/pull/2973#pullrequestreview-2458214556 . This is mostly applicable if your code is using `test_inside_container` and the changes would be minimal, but please rebase on main to be confirm. Thanks.

**Comment by ntkm61027** [comment]
Hi @YJDoc2
I'm new here and would like to try `process_rlimits_fail`. Would you mind assigning it to me?

**Comment by utam0k** [comment]
@ntkm61027 I've assigned you!

**Comment by moz-sec** [comment]
@YJDoc2 
Now that `linux_cgroups_relative_network` has been merged, is it possible to have `linux_uid_mappings` assigned to me?

**Comment by utam0k** [comment]
> [@YJDoc2](https://github.com/YJDoc2) Now that `linux_cgroups_relative_network` has been merged, is it possible to have `linux_uid_mappings` assigned to me?

Sure! Thanks a lot 🙏

**Comment by you-matsuura** [comment]
@utam0k 
I would like to be assigned to `mounts`
Could you please assign it to me?

**Comment by utam0k** [comment]
@you-matsuura Hi, thanks for your comment. I'll assign you.

**Comment by saku3** [comment]
I would appreciate it if you could update the list of tests in the issue for clarity.

The following tests have already been implemented, so please mark them as completed (checked):

- [kill](https://github.com/youki-dev/youki/pull/2996)
- [linux_masked_paths](https://github.com/youki-dev/youki/pull/2950)
- [linux_rootfs_propagation](https://github.com/youki-dev/youki/pull/3024)
- [linux_uid_mappings](https://github.com/youki-dev/youki/pull/3161)
- [process_capabilities_fail](https://github.com/youki-dev/youki/pull/3010)
- [process_oom_score_adj](https://github.com/youki-dev/youki/pull/2987)
- [process_rlimits_fail](https://github.com/youki-dev/youki/pull/3051)
- [root_readonly_true](https://github.com/youki-dev/youki/pull/2976)

**Comment by YJDoc2** [comment]
Hey @saku3 Thanks for listing this and linking all the PRs 🙏  I hvae updated the list :+1:

**Comment by YamasouA** [comment]
@YJDoc2 @utam0k 
Can I try `misc_props`?

**Comment by utam0k** [comment]
@YamasouA Sure!

**Comment by YamasouA** [comment]
@utam0k @YJDoc2 
I also want to try the `killsig` test.

**Comment by YJDoc2** [comment]
@YamasouA sure, I have assigned you

**Comment by HukuKaich0u** [comment]
Hi @utam0k .
I would like to try this **_linux_process_apparmor_profile_**. Please assign me this test.🙇‍♂️

**Comment by utam0k** [comment]
@HukuKaich0u Sure. I'll assign you.

**Comment by cody-herbst** [comment]
Hi @utam0k .

I would like to take a stab at linux_ns_path if that's ok.

**Comment by utam0k** [comment]
> Hi [@utam0k](https://github.com/utam0k) .
> 
> I would like to take a stab at linux_ns_path if that's ok.

Sure! I've assigned it to you.

**Comment by cody-herbst** [comment]
Hi @utam0k .

I have a PR up for linux_ns_path. Can you assign me linux_ns_nopath and linux_ns_path_type? I'll start on those once linux_ns_path is approved so I know of any gotchas going forward. In the mean time will continue learning the main youki application

**Comment by utam0k** [comment]
@cody-herbst Sure! Thanks.

**Comment by fspv** [comment]
Hey @utam0k ! Can I claim poststart? I have already created a draft PR https://github.com/youki-dev/youki/pull/3292

**Comment by YJDoc2** [comment]
> Hey @utam0k ! Can I claim poststart? I have already created a draft PR https://github.com/youki-dev/youki/pull/3292

Assigned :+1:

**Comment by fspv** [comment]
@utam0k @YJDoc2 the poststart is now landed. I picked up the poststart_fail here https://github.com/youki-dev/youki/pull/3313

**Comment by oneplus1000** [comment]
Hello @utam0k, @YJDoc2 — could you assign kill_no_effect to me? I’ve already created the PR.

**Comment by utam0k** [comment]
> Hello [@utam0k](https://github.com/utam0k), [@YJDoc2](https://github.com/YJDoc2) — could you assign kill_no_effect to me? I’ve already created the PR.

Sure!

**Comment by fspv** [comment]
Working on prestart here https://github.com/youki-dev/youki/pull/3382 Please assign to me

**Comment by saku3** [comment]
>Working on prestart here https://github.com/youki-dev/youki/pull/3382 Please assign to me

Assigned

**Comment by donkomura** [comment]
Is anyone assigned to poststop? If not, I'd like to work on it, please assign me.

**Comment by saku3** [comment]
> Is anyone assigned to poststop? If not, I'd like to work on it, please assign me.

Assigned

**Comment by utam0k** [comment]
@fspv @donkomura Thanks for your rising hands!!

**Comment by fspv** [comment]
Hey! I have created a PR for the prestart_fail hook https://github.com/youki-dev/youki/pull/3406, please assign to me.

Additionally, I have created a draft PR for the poststop_fail hook, which I believe contains a bug (same as in runc), so please take a look, because I'm not sure what approach to take there https://github.com/youki-dev/youki/pull/3407 (sorry for breaking "Please only work at one test at a time", just wanted to highlight this bug earlier)

**Comment by saku3** [comment]
I think the mounts tests are important, so I’ll take care of them.

I spoke with @you-matsuura  separately from this thread, and I’ll be taking over from here.

**Comment by aayush0325** [comment]
Hi, can I take over one of the unassigned tests? I see that `hooks_stdin` and `process_capabilities` are not assigned to anyone

**Comment by saku3** [comment]
@aayush0325 
Sure! Either one is fine. Please choose whichever you prefer, and I’ll assign it to you.

**Comment by aayush0325** [comment]
I'll work on `process_capabilities` then, thanks @saku3!

**Comment by saku3** [comment]
>I'll work on process_capabilities

Assigned.

**Comment by ITSMERNB** [comment]
Hi! I'd like to work on the `state` validation test port if it's still available.

I'll keep the scope to this single test, follow the current contest test framework/style, and use the upstream OCI runtime-tools `validation/state` test as the behavioral reference.

**Comment by saku3** [comment]
> Hi! I'd like to work on the state validation test port if it's still available.

Assigned.

**Comment by ITSMERNB** [comment]
Thanks for assigning it, @saku3. PR #3697 is up and the Linux/OCI CI has run successfully for the relevant `oci-validation-rust (x86_64, gnu)` path along with the main build/test checks. I’ll keep the scope to `state` only and wait for this PR to be reviewed/merged before taking another test.

**Comment by aayush0325** [comment]
Hi, can i work on `linux_cgroups_relative_blkio`? @saku3

**Comment by saku3** [comment]
As far as I can tell, `linux_cgroups_relative_blkio` in runtime-tools is a test for cgroup v1.

It makes sense to add cgroup v2 tests for the blkio.

**Comment by aayush0325** [comment]
> It makes sense to add cgroup v2 tests for the blkio.

does that go in `tests/contest/contest/src/tests/cgroups`?

**Comment by karthik20066002** [comment]
Hi, are any of the issues still open to be taken? I see that the listings might be out of date. @saku3

**Comment by saku3** [comment]
@karthik20066002 

Thank you for letting me know.

I'll update the list.

I'd particularly like to see the following tests implemented:

* `validateDefaultDevices`
* `validateDefaultSymlinks`
* `validateDefaultFS`

These tests verify that the default devices, symlinks, and filesystem entries are correctly set up when a container is created.

https://github.com/opencontainers/runtime-tools/blob/master/cmd/runtimetest/main.go#L50

**Comment by karthik20066002** [comment]
I'll take `ValidateDefaultSymlinks` then, if you don't mind.

**Comment by saku3** [comment]
@karthik20066002 
Thanks.  I assigned you.

**Comment by karthik20066002** [comment]
@saku3 please review whenever able, thank you!
